flink-runtime/src/main/java/org/apache/flink/runtime/util/ZooKeeperUtils.java

```diff
@@ -354,13 +362,23 @@ public static ZooKeeperCheckpointIDCounter createCheckpointIDCounter(
 			Configuration configuration,
 			String prefix) throws IOException {
 
+		return new FileSystemStateStorageHelper<T>(getClusterHighAvailabilityStoragePath(configuration), prefix);
+	}
+
+	/**
+	 * Get high availability storage path of current flink cluster.
+	 * @param configuration {@link Configuration} object
+	 * @return high availability storage path of current flink cluster
+	 */
+	private static String getClusterHighAvailabilityStoragePath(Configuration configuration) {
 		String rootPath = configuration.getValue(HighAvailabilityOptions.HA_STORAGE_PATH);
 
 		if (rootPath == null || StringUtils.isBlank(rootPath)) {
 			throw new IllegalConfigurationException("Missing high-availability storage path for metadata." +
 					" Specify via configuration key '" + HighAvailabilityOptions.HA_STORAGE_PATH + "'.");
 		} else {
-			return new FileSystemStateStorageHelper<T>(rootPath, prefix);
+			final String clusterId = configuration.getValue(HighAvailabilityOptions.HA_CLUSTER_ID);
```

Comment:
What if `HighAvailabilityOptions.HA_CLUSTER_ID` configured wrongly? Shall we perform a runtime checker to guard `clusterId` is valid?

Reply:
What do you mean configured wrongly? In standalone mode, it is configured by user. In Yarn/mesos mode, it will be set automatically. Also it have default value and will not be null.

Reply:
Yes your right. It's my mistake.

Reply:
I think it is done partially because Till check the clusterId is not empty or whitespace-only. I remember my concern here previous that if the path concatenate is not a valid path it might fail later.